### ISSUE TITLE
Remove failure with new session.

### ISSUE THREAD

**Issue by jbkoh** [davebshow/goblin#60]
Hi,

(I use JanusGraph 0.1.1, Python 3.6.1, and Goblin 2.0.0 (from pip) and saw an error.)
It fails at removing a vertex with a new session. Here's the example code:

```python
import asyncio
from goblin import Goblin
from goblin.properties import Property, String, Boolean
from goblin.element import Vertex, Edge
import goblin

def get_hashable_id(val):
    #Use the value "as-is" by default.
    result = val
    if isinstance(val, dict) and "@type" in val and "@value" in val:
        if val["@type"] == "janusgraph:RelationIdentifier":
            result = val["@value"]["value"]
    return result

class Person(Vertex):
    name = Property(String)

async def gen_jane(app):
    session = await app.session()
    jane = Person()
    jane.name = 'JaneDoe'
    session.add(jane)
    await session.flush()

async def del_jane(app):
    session = await app.session()
    trav = session.traversal(Person).has('name', 'JaneDoe')
    jane = (await trav.toList())[0]
    await session.remove_vertex(jane)
    

loop = asyncio.get_event_loop()
app = loop.run_until_complete(Goblin.open(loop,
    get_hashable_id=get_hashable_id))
app.config_from_file('remote.yaml')
app.register(Person)

loop.run_until_complete(gen_jane(app))
loop.run_until_complete(del_jane(app))
```

The above code produces 

---
Traceback (most recent call last):
  File "jane_test.py", line 40, in <module>
    loop.run_until_complete(del_jane(app))
  File "/home/jbkoh/anaconda3/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/home/jbkoh/anaconda3/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/home/jbkoh/anaconda3/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "jane_test.py", line 30, in del_jane
    await session.remove_vertex(jane)
  File "/home/jbkoh/anaconda3/lib/python3.5/site-packages/goblin/session.py", line 281, in remove_vertex
    vertex = self.current.pop(hashable_id)
  File "/home/jbkoh/anaconda3/lib/python3.5/weakref.py", line 240, in pop
    o = self.data.pop(key)()
KeyError: 20728

---

I checked that the vertex is actually removed in JanusGraph and the _id was found correctly as 20728 for jane.

Is this an expected behavior? How can I remove a vertex with a query? (I would like to use it as a normal CRUD database.)

**Comment by davebshow** [comment]
Hi @jbkoh. I am currently out of the country with little to no internet connection. I will be back this weekend and will look into this issue.

**Comment by jbkoh** [comment]
@davebshow that will work. Thank you!

**Comment by davebshow** [comment]
This is a bug introduced in the last refactor. Working on a fix now.

**Comment by davebshow** [comment]
This should be fixed on master. Please install from github and verify.

```
$ pip install git+https://github.com/davebshow/goblin.git
```

**Comment by davebshow** [comment]
@jbkoh Is this working for you now? I am preparing for a release and would like to be able to close this issue.

**Comment by jbkoh** [comment]
Sorry I couldn't find time to revisit this issue. I will try tonight and let you know. (Or I can close after test it by myself.)

**Comment by jbkoh** [comment]
It works perfectly right now. Thanks for the fix.

**Comment by StanYaha** [comment]
@davebshow 
hi，my golblin version is 2.0.0, but i encounter the same problem.
Have you fixed this bug in goblin(2.0.0)?

I failed when i
```
  pip install git+https://github.com/davebshow/goblin.git
```
```
[root@Master graph]# pip install git+https://github.com/davebshow/goblin.git
Collecting git+https://github.com/davebshow/goblin.git
  Cloning https://github.com/davebshow/goblin.git to /tmp/pip-rvk8v5as-build
  Error [Errno 2] No such file or directory: 'git' while executing command git clone -q https://github.com/davebshow/goblin.git /tmp/pip-rvk8v5as-build
Cannot find command 'git'
```
```
Traceback (most recent call last):
  File "/usr/local/python36/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/python36/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/python36/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/python36/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/usr/local/python36/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/python36/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "stan.py", line 109, in del_vertex
    loop.run_until_complete(delvertex(id_number))
  File "/usr/local/python36/lib/python3.6/asyncio/base_events.py", line 466, in run_until_complete
    return future.result()
  File "stan.py", line 80, in delvertex
    await session.remove_vertex(vertex)
  File "/usr/local/python36/lib/python3.6/site-packages/goblin/session.py", line 281, in remove_vertex
    vertex = self.current.pop(hashable_id)
  File "/usr/local/python36/lib/python3.6/weakref.py", line 240, in pop
    o = self.data.pop(key)()
KeyError: 3338352
```

**Comment by davebshow** [comment]
This fix has not yet been released. You can install Goblin from github for the time being:

```
$ pip install git+https://github.com/davebshow/goblin.git
```
